packages/flutter_test/lib/src/controller.dart

```diff
@@ -23,6 +23,144 @@ const double kDragSlopDefault = 20.0;
 
 const String _defaultPlatform = kIsWeb ? 'web' : 'android';
 
+/// Class that programatically interacts with the Semantics tree.
+class SemanticsController {
+  /// Creates a SemanticsController that uses the given binding.
+  SemanticsController(this.binding);
+
+  static final int _scrollingActions =
+    SemanticsAction.scrollUp.index |
+    SemanticsAction.scrollDown.index |
+    SemanticsAction.scrollLeft.index |
+    SemanticsAction.scrollRight.index;
+
+  static final int _importantFlags =
```

Comment:
TBH, it's something I'm worried about as well. This list was created based on the `FOCUSABLE_FLAGS` property in the engine repo's `AccessibilityBridge.java#AccessibilityBridge` class. The iOS side is more based on the `actions` instead of the `flags`.

I debated leaving out the flags entirely since they seem to overlap with the "has any non-scrolling action" check, but I added them in because I could see a use case where someone wanted something that could _not_ be actioned upon, but still should be described by a flag. For example, a page that has a read-only mode and edit mode. The edit mode would have actionable widgets, but the read-only one could just have widgets that describe the current state as "checked". If this isn't a use case we're worried about, I'd lean towards removing the flags check, since the "has any non-scrolling action" check is more future proof.

If we're going to keep the flag check as well, then based off of current examples, I'd say I could add to the comments on `SemanticsFlag` that this value needs to be updated as well as those in the engine, as that's currently where the engine's `SemanticsFlag` values are "enforced".

Reply:
I added a comment on this field to point to the `AccessibilityBridge.java` class that it's based off of. I also added to the comments in `semantics.dart#SemanticsFlag` where it talks about what needs to be updated when SemanticsFlag is changed/added to. Those comments should drive developers to this field as well as those they'd have to update already.

Reply:
I don't think the comment in `semantics.dart#SemanticsFlag` made it into this PR?

Reply:
Well, technically no it didn't. Turns out `semantics.dart#SemanticsFlag` is actually defined in the _engine_ repo. I only found that out after I couldn't find the change I had made locally, then noticed that the file I changed was actually in my cache, not my flutter repo.

Now that we're close to wrapping this PR up, here's the [relevant PR on the engine side](https://github.com/flutter/engine/pull/37046).